tests/helpers/testers.py

```diff
@@ -114,18 +114,24 @@ def _class_test(
     metric = pickle.loads(pickled_metric)
 
     for i in range(rank, NUM_BATCHES, worldsize):
-        batch_result = metric(preds[i], target[i])
+        b_kwargs_update = {k: v[i] for k, v in kwargs_update.items()}
+
+        batch_result = metric(preds[i], target[i], **b_kwargs_update)
 
         if metric.dist_sync_on_step:
             if rank == 0:
                 ddp_preds = torch.cat([preds[i + r] for r in range(worldsize)])
                 ddp_target = torch.cat([target[i + r] for r in range(worldsize)])
-                sk_batch_result = sk_metric(ddp_preds, ddp_target)
+                ddp_kwargs_upd = {
+                    k: torch.cat([v[i + r] for r in range(worldsize)]) for k, v in b_kwargs_update.items()
```

Comment:
same as above

Reply:
update in https://github.com/PyTorchLightning/metrics/pull/137/commits/86e4015cb0bf85bd10c7f860ada0c8a0c93124d5